android/lib/src/main/java/org/vosk/android/SpeechService.java

```diff
@@ -314,15 +330,21 @@ class TranscriptionThread extends Thread {
 
         RecognitionListener recognitionListener;
 
+        private boolean isToResetRecognizer = false;
+
         TranscriptionThread(int queueSize, RecognitionListener recognitionListener) {
             this.recordingChunksQueue = new ArrayBlockingQueue<>(queueSize);
             this.recognitionListener = recognitionListener;
         }
 
+        private void isToResetRecognizer() {
+            this.isToResetRecognizer = true;
+        }
+
         @Override
         public void run() {
 
-            while (!isInterrupted()) {
+            while (!isInterrupted() && !isToResetRecognizer) {
```

Comment:
why adding the flag to reset the recognizer here?
* Do we want to reset the queue and exit the while loop? If so, we shouldn't put the variable back to false when we reset the queue.
* Do we want to reset the queue but keep in the loop? Then we shouldn't have this condition guarding the while